python-sdk/src/astro/databases/google/bigquery.py

```diff
@@ -60,6 +60,9 @@
 }
 BIGQUERY_WRITE_DISPOSITION = {"replace": "WRITE_TRUNCATE", "append": "WRITE_APPEND"}
 
+NATIVE_AUTODETECT_SCHEMA_SUPPORTED_FILE_TYPES = {FileType.CSV, FileType.NDJSON}
+NATIVE_AUTODETECT_SCHEMA_SUPPORTED_FILE_LOCATIONS = {FileLocation.GS}
```

Comment:
@feluelle How would we represent cases like:
S3 - Parquet
GCS - CSV, JSON, Parquet

Reply:
> GCS - CSV, JSON, Parquet

`GS` is GCS; CSV & JSON is supported; For parquet the native inference within bigquery will be used automatically. See comment https://github.com/astronomer/astro-sdk/pull/780#discussion_r967004658

Reply:
> S3 - Parquet

I guess, there is not native path available which means it will use pandas instead.

Reply:
So currently what this mapping implies is that for every supported location(GCS/S3) we support all the filetypes - CVS/JSON, right? but is it the case always? 

```
is_file_type_supported = (
    file.type.name in NATIVE_AUTODETECT_SCHEMA_SUPPORTED_FILE_TYPES
)
is_file_location_supported = (
    file.location.location_type
    in NATIVE_AUTODETECT_SCHEMA_SUPPORTED_FILE_LOCATIONS
)
```
Would be a good idea to check for location and then support fileType by that location? Something like - 
```
{
"GCS": [CSV, NDJSON],
"S3": [PARQUET]  # just an example. 
}
```

Reply:
but we do - in the `return` statement:
```python
return is_file_type_supported and is_file_location_supported
```

Reply:
ah, okay I got what you mean now. Yes, this could make sense. I just used the same functionality we use for file loading. Is file loading different from schema autodetection? 🤔

Reply:
@feluelle No, I guess not. We should change that as well. I just realized that we can do that in a separate PR.

Reply:
@utkarsharma2 Can you create a separate issue on the separate PR you are talking about?

Reply:
@kaxil I think @feluelle is truncating the table: https://github.com/astronomer/astro-sdk/blob/main/python-sdk/src/astro/databases/google/bigquery.py#L279

The test only needs fix

Reply:
Added - https://github.com/astronomer/astro-sdk/issues/853